mango/config/base_config.py

```diff
@@ -0,0 +1,169 @@
+"""
+This file contains the classes that are used to parse the configuration files
+"""
+import os
+import warnings
+from configparser import ConfigParser, NoOptionError, NoSectionError
+from typing import Union
+
+
+class ConfigParameter:
+    """
+    This class represents a parameter in a configuration file with INI File syntax.
+    Each parameter has a name (their section gets parsed from the file), value type, default value, list of valid values
+    and a secondary type (if the value type is a list).
+    """
+
+    def __init__(
+        self,
+        name: str,
+        value_type: callable,
+        default: Union[int, float, bool, str, list] = None,
```

Comment:
We may want to switch to the List typing type to avoid any errors in future versions.